api/v1beta1/cluster_types.go

```diff
@@ -184,6 +196,32 @@ type MachineDeploymentTopology struct {
 	Variables *MachineDeploymentVariables `json:"variables,omitempty"`
 }
 
+// MachineHealthCheckTopology defines a MachineHealthCheck for a group of machines.
+// MachineHealthCheckTopology can be used to enable or disable the creation of a
+// MachineHealthCheck for the target machines.
+// If specified, it overrides the values provided in the ClusterClass.
+type MachineHealthCheckTopology struct {
+	// Enable controls if a MachineHealthCheck should be creates for the target machines.
+	// If false:
+	//		No MachineHealthCheck will be created.
+	// If not set:
+	//		A MachineHealthCheck will be created if it is defined here or in the associated ClusterClass.
+	//		If not MachineHealthCheck if defined then none will be created.
+	// If true:
+	//		A MachineHealthCheck is guaranteed to be created. Cluster validation will block if `enable` is
+	//		true and no MachineHealthCheck definition is available.
+	// +optional
+	Enable *bool `json:"enable,omitempty"`
+
+	// Spec defines a MachineHealthCheck for a group of machines.
+	// TODO: Rename this to something better.
+	// NOTE: Nesting the MHC definition here so that it can be used to determine if an MHC is defined in the Cluster topology.
+	//       Alternative would be to check each field to see if MHC is defined in topology.
+	//
+	// +optional
+	Spec *MachineHealthCheckClass `json:"spec,omitempty"`
```

Comment:
Added this nesting here becasue it can be used to check if a MachineHealthCheck is not defined in the Cluster topology. It makes consuming this type easier.

Reply:
Any suggestions for better names are welcome :)